### PR TITLE
feat(contracts): add contracts crate, remove from types

### PR DESCRIPTION
**Pull request by dancoombs**

### CODE REVIEW

File: crates/contracts/build.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        .get("object")
+        .unwrap()
+        .as_str()
+        .unwrap();
+    fs::write(
+        "contracts/out/v0_7/EntryPointSimulations.sol/EntryPointSimulations_deployedBytecode.txt",
+        bytecode,
+    )?;
+
+    let json =
```
Comment (dancoombs) [inline crates/contracts/build.rs:65]: TODO: DRY

File: crates/contracts/src/v0_6.rs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+//
+// You should have received a copy of the GNU General Public License along with Rundler.
+// If not, see https://www.gnu.org/licenses/.
+
+// Contracts from https://github.com/eth-infinitism/account-abstraction/tree/releases/v0.6/contracts
+
+use alloy_primitives::Bytes;
+use alloy_sol_macro::sol;
+
+sol! {
```
Comment (dancoombs) [inline crates/contracts/src/v0_6.rs:19]: The reason for using this macro and shoving all of the contract interfaces into the same macro is that alloy currently doesn't deduplicate shared types across invocations to `sol!`, and `sol!` only supports 1 file import.

Will be filing a feature request with them when I get around it.

Much better to deal with this ugliness here than to have 5 different `UserOperation` structs compiled from different contracts.

File: crates/contracts/src/v0_7.rs [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    GetBalances,
+    "contracts/out/v0_7/GetBalances.sol/GetBalances.json"
+);
+
+// EntryPointSimulations deployed bytecode
```
Comment (dancoombs) [inline crates/contracts/src/v0_7.rs:246]: We need `EntryPointSimulations` in the large `sol!` macro above for types sharing. However there is no good way to get the bytecode in that (other than pasting hex into the macro). This, combined with the build script, pulls the compiled bytecode from foundry into this file.

**Review by 0xfourzerofour — APPROVED** [review]
LGTM

**Review by andysim3d — APPROVED** [review]

**Pull request merged**